addons/website/models/website_menu.py

```diff
@@ -158,3 +161,11 @@ def replace_id(old_id, new_id):
             menu_id.write(menu)
 
         return True
+
+    @api.multi
+    def toggle_mega_menu(self):
+        if self.mega_menu_content:
+            default_mega_menu = False
+        else:
+            default_mega_menu = self.env.ref('website.default_mega_menu').read()[0]['arch_base']
+        self.write({'mega_menu_content': default_mega_menu})
```

Comment:
In fact I would not use a temporary `default_mega_menu` variable, do directly `self.mega_menu_content = ...` in the if and else.